### PR TITLE
Fix `StyleManager` default style updates.

### LINKED ISSUE #3917: [Bug]: `DayStyle` and `NightStyle` updates should only affect one specific platform.
### Mapbox Navigation SDK version

main

### Steps to reproduce

At this point `StyleManager` is used across multiple platforms (iOS and CarPlay), at the same time `DayStyle` and `NightStyle` contain code that changes appearance globally. This leads to issues in components that are shared between these platforms (e.g. `LaneView`) whenever either iOS or CarPlay trigger style update.

In scope of this bug fix, `DayStyle` and `NightStyle` should be adapted for the independent style changes to affect only individual platform.

### Expected behavior

Style update on one platform should not affect another platform.

### Actual behavior

Style update on one platform affects another platform.

### Is this a one-time issue or a repeatable issue?

repeatable

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

Fixes #3917.

PR adds trait collection check in default `DayStyle` and `NightStyle` to make sure that style is not updated for iOS in case if CarPlay style changes (and vice versa).

### CODE REVIEW

File: Sources/MapboxNavigation/NightStyle.swift [hunk 1]
```diff
@@ -20,117 +20,184 @@ open class NightStyle: DayStyle {
...
-        StepsTableHeaderView.appearance().normalTextColor = #colorLiteral(red: 0.9996390939, green: 1, blue: 0.9997561574, alpha: 1)
-        UILabel.appearance(whenContainedInInstancesOf: [FeedbackViewController.self]).backgroundColor = .black
-        UILabel.appearance(whenContainedInInstancesOf: [FeedbackViewController.self]).textColor = .white
-        FeedbackStyleView.appearance(whenContainedInInstancesOf: [FeedbackViewController.self]).backgroundColor = .black
-        FeedbackCollectionView.appearance().backgroundColor = .black
-        FeedbackCollectionView.appearance().cellColor = .white
-        FeedbackSubtypeCollectionViewCell.appearance().normalCircleColor = .black
-        FeedbackSubtypeCollectionViewCell.appearance().normalCircleOutlineColor = .lightText
-        
-        WayNameView.appearance().borderColor = #colorLiteral(red: 0.2802129388, green: 0.3988235593, blue: 0.5260632038, alpha: 1)
```
Comment (MaximAlien) [inline Sources/MapboxNavigation/NightStyle.swift:119]: Appearance changes should not affect all views across all windows. Changes should be applied only for specific user interface idiom (at this point it's either iOS or CarPlay).

**Comment by MaximAlien** [comment]
> @MaximAlien How this change affects users that override Style with own customizations? Looks like we also need to update docs and examples to reflect this new recommended way of changing style. We should mention the change in the changelog and how to migrate.

Users are fully responsible for their own styles and current functionality doesn't interfere with custom styles. It's probably a good idea to mention that global appearance updates of certain views that are used on CarPlay will affect all platforms and trait collections should be used to create individual style changes. I'll create separate ticket to update public documentation for this. Will update changelog as well.

**Review by 1ec5 — APPROVED** [review]
I guess the changelog entry could recommend taking a look at the changes we made in DayStyle.swift, which are quite numerous. The custom style example would be updated, but it isn’t publicly version controlled, so it would be more difficult for developers to follow along by looking at it.

File: Tests/MapboxNavigationTests/CarPlayCompassViewSnapshotTests.swift [hunk 2]
```diff
@@ -4,7 +4,15 @@ import SnapshotTesting
 import TestHelper
 @testable import MapboxNavigation
 
+class CarPlayCompassViewMock: CarPlayCompassView {
+    
+    override var traitCollection: UITraitCollection {
+        return UITraitCollection(userInterfaceIdiom: .carPlay)
```
Comment (MaximAlien) [inline Tests/MapboxNavigationTests/CarPlayCompassViewSnapshotTests.swift:10]: This is needed to make `Style` think that this view was created on CarPlay.

**Comment by MaximAlien** [comment]
Cherry-picked to `release-v2.6` branch in https://github.com/mapbox/mapbox-navigation-ios/commit/7e5db91a49dda60e855240563bb0834afa3e7bdd.

**Pull request merged**